Makefile

```diff
@@ -0,0 +1,23 @@
+.PHONY: build
```

Comment:
I'm not sure of the point of the Makefile. Is it just so people who want to use the C library but aren't familiar with cargo can get started with the library easily?

If we do decide to add the Makefile, please also add a note to the readme, or a comment in this file, explaining that it's for the C library specifically. Otherwise I think it's a bit confusing to have a Makefile in a Rust project.

Reply:
The main reason for the `Makefile` is to have targets for running tests and linters. The `build` target was just an extra.

If you want I can do one of the following things:
- Remove the `build` target
- Rename the `build` target to `build-capi` or something better. Then update the `README`

BTW, I didn't personally find odd to have a `Makefile` inside of a Rust project. That's something I've seen in multiple places (along with more hipster [`just`](https://github.com/casey/just) :smile: )

Reply:
I just don't really understand what the Makefile adds... typing `make fmt` is not any easier than typing `cargo fmt`.

But it's not hurting anything, so I don't mind having it if you prefer. But yes, let's at least rename `build` to `capi`.

Reply:
I've done the rename. As for the added value, I forgot to mention the main benefit is the ability to run `make test` and have code formatting checked, linter run and documentation checks too.

All of that is consistent inside of the local environment and inside of the GitHub Action (since it relies on the same make target)